pyrmq/consumer.py

```diff
@@ -131,7 +154,7 @@ def close(self) -> None:
         """
         Manually closes a connection to RabbitMQ. Useful for debugging and tests.
         """
-        self.connection.close()
+        self.thread.join(0.1)
```

Comment:
?

Reply:
So there's no way to manually close a thread in Python, right? But rerunning the tests results in flaky behavior which I think stems from opened connections by pika. Will have to investigate further. In any case, one way to "stop" it is by joining it back to the main thread. This essentially ends the consumption and allows better cleanup per test. I'm still thinking whether this should be part of the global teardown but I figure it's better to leave here so devs can see that multiple consumers should be closed to maintain consistency in dealing with a single queue or exchange.

Reply:
Why 0.1?

Reply:
It's the timeout value. Could be anything really. Just tried 0.1 seconds to make it resolve as fast as possible. There might be a better way.

Reply:
Okay, we can leave it for now. Besides, we're only using this for tests.